src/client-side-metrics/gcp-metrics-handler.ts

```diff
@@ -49,8 +55,8 @@ interface MetricsInstruments {
  * associating them with relevant attributes for detailed analysis in Cloud Monitoring.
  */
 export class GCPMetricsHandler implements IMetricsHandler {
-  private otelInstruments?: MetricsInstruments;
   private exporter: PushMetricExporter;
+  private projectToInstruments: {[projectId: string]: MetricsInstruments};
```

Comment:
This is supposed to be global shared state, right? Can't we make this a static variable to be shared across all instances?

I find it confusing that this looks like an instance variable, but we actually expect it to point to a shared singleton in practice.

Reply:
Yes. This is a really good idea and it solves the problem of resetting the map in the tests too.